### ISSUE TITLE
Empty Param value doesn't propagate to JSON

### ISSUE THREAD

**Issue by BasPH** [apache/airflow#31869]
### Apache Airflow version

2.6.1

### What happened

I'm using `array` Params and sometimes want to provide no values.

The Param is configured with `type=["array", "null"]` and a default value (e.g. `["a", "b", "c"]`). I would assume emptying the Param field would set an empty array or null in the generated configuration JSON, but instead Airflow sets the default value. I have to manually modify the generated configuration JSON to submit an empty array.

Normal usage:
![image](https://github.com/apache/airflow/assets/6249654/61b24ab9-ad83-4b6d-9fcb-b47a55e88356)

Clearing the param field sets the default value:
![image](https://github.com/apache/airflow/assets/6249654/ee6110e9-2829-4c47-8862-31503c7abdac)

I have to manually modify the generated configuration JSON:
![image](https://github.com/apache/airflow/assets/6249654/bb4be772-0bb9-4be5-b2f8-280c8f5ad917)

### What you think should happen instead

I expect Airflow to set an empty array or null value when a Param field is left empty (and `null` is set as Param type).

### How to reproduce

```python
import datetime

from airflow.decorators import dag, task
from airflow.exceptions import AirflowSkipException
from airflow.models import Param


@dag(
    schedule=None,
    start_date=datetime.datetime(2023, 1, 1),
    params={
        "demo": Param(
            default=["a", "b", "c"],
            type=["array", "null"],
        )
    },
)
def dagrun_conf():
    @task
    def print_params(params=None):
        if params["demo"]:
            print(params["demo"])
        else:
            raise AirflowSkipException("Nothing to do here")

    print_params()


dagrun_conf()

```

### Operating System

N/A

### Deployment

Astronomer

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by jens-scheffler-bosch** [comment]
Can confirm this. Though I assume it is in the WebUI Part and not in the core. I'll take a look for a fix if not somebody else is faster.

**Comment by potiuk** [comment]
Assigned you @jens-scheffler-bosch

**Comment by jens-scheffler-bosch** [comment]
This bug was introduced by a fix in PR https://github.com/apache/airflow/pull/31078/files#diff-74480522dd1ea4a8a51b8f9c6355372f5384440a911b2baf8d0a75786cc7c420R49 and used the default/original value if array is empty. So it started appearing in 2.6.1 and also is like this in 2.6.2 if we cut the release.
It is fixed on main already implicitly by PR #31441
As 2.6.2 is in the final stage and the problem is rather low severity I would leave it. I could offer a 1-line fix for a future 2.6.3 maintenance. Will this be needed or shall we close it assuming that 2.7.0 has fixed the problem already?

**Comment by jens-scheffler-bosch** [comment]
Hi @BasPH after consultation with the experts I now understand the fix and release process. The fix included in next 2.7++ can not be directly cherry-picked and a PR on the maintenance branch for 2.6 is not the way of working.
Therefore to keep complexity low for this bug I'd propose to keep the fix in 2.7++ and not fix it in 2.6 maintenance. Let me know if you see severity higher, then we would need to revert and modify the feature in 2.7 which includes the fix in order to back-port this to 2.6 line.

**Comment by BasPH** [comment]
2.7.0 is fine!

**Comment by jens-scheffler-bosch** [comment]
closed by #31441
